src/core/lib/request.js

```diff
@@ -58,7 +57,10 @@ function wait(retryDelay) {
     });
 }
 
-function fetchRetry(url, headers, fetchOptions, resolve, reject, retryDelay = 300, retryLimit = 5) {
+function fetchRetry(stack, queryParams, fetchOptions, resolve, reject, retryDelay = 300, retryLimit = 5) {
+    let requestParams = stack.requestParams,
```

Comment:
Are these being reassigned, if not let's use const?
Consider removing var from the below line too